src/main/java/uk/gov/ons/ctp/integration/rhsvc/service/CaseService.java

```diff
@@ -14,13 +14,16 @@
 public interface CaseService {
 
   /**
-   * Retrieve the data relating to non-HI latest valid Case by address UPRN
+   * Retrieve case(s) which have a sample attribute matching the supplied key/value.
    *
-   * @param uprn of address for which latest valid case details are requested
-   * @return Case details for address UPRN
+   * @param attributeKey is the name of the attribute to search on.
+   * @param attributeValue is the value that the search attribute must equal for a case to be
+   *     returned.
+   * @return a List containing matching cases. If no cases then the list will be empty.
    * @throws CTPException if anything went wrong.
    */
-  CaseDTO getLatestValidCaseByUPRN(final UniquePropertyReferenceNumber uprn) throws CTPException;
+  List<CaseDTO> readCaseUpdateBySampleAttribute(
```

Comment:
This method name suggests just returning one case. Consider prefix of `listCases` or `findCases` or something

Reply:
Have renamed to findCasesBySampleAttribute()